### ISSUE TITLE
Causal clustering is not configurable with Docker service name (swarm mode)

### ISSUE THREAD

**Issue by jarellm** [neo4j/docker-neo4j#87]
I'm currently testing out the causal clustering feature in Docker swarm mode primarily with a docker-compose file and the command: 

`docker stack up -c compose-neo4j-master.yml neo4j`

Ideally, the most concise "swarm mode way" to deploy the 3 core cluster nodes would be this:

```
services:
  neo4jcore:
     - NEO4J_dbms_mode=CORE
     - NEO4J_causalClustering_expectedCoreClusterSize=3
     - NEO4J_causalClustering_initialDiscoveryMembers=neo4jcore:5000
     deploy:
        replicas: 3
```

The  `neo4jcore:5000` part leverages the built-in DNS resolution and routing mesh within the neo4jcore service, so I wouldn't have to know the hostname/IP of each task container spawned by Docker.

However, running this configuration stops at

`Attempting to connect to the other cluster members before continuing..`

I haven't dug in the code, but I'm assuming either the expected count will never reach 3 because only 1 host was declared OR the load balancing provided by the docker swarm is getting in the way for core member nodes to properly do handshakes.

What did work for me is to do something similar to your test file [causal-cluster-compose.yml](https://github.com/neo4j/docker-neo4j/blob/master/test/causal-cluster-compose.yml) and manually spread the services to different nodes.

```
services:
  neo4jcore1:
     - NEO4J_dbms_mode=CORE
     - NEO4J_causalClustering_expectedCoreClusterSize=3
     - NEO4J_causalClustering_initialDiscoveryMembers=neo4jcore1:5000,neo4jcore2:5000,neo4jcore3:5000
     deploy:
        replicas: 1
        placement:
          constraints:
             - node.hostname=node1
  neo4jcore2:
     - NEO4J_dbms_mode=CORE
     - NEO4J_causalClustering_expectedCoreClusterSize=3
     - NEO4J_causalClustering_initialDiscoveryMembers=neo4jcore1:5000, neo4jcore2:5000,neo4jcore3:5000
     deploy:
        replicas: 1
        placement:
          constraints:
             - node.hostname=node2
  neo4jcore3:
     - NEO4J_dbms_mode=CORE
     - NEO4J_causalClustering_expectedCoreClusterSize=3
     - NEO4J_causalClustering_initialDiscoveryMembers=neo4jcore1:5000, neo4jcore2:5000,neo4jcore3:5000
     deploy:
        replicas: 1
        placement:
          constraints:
             - node.hostname=node3
```

It would be great if neo4j/neo4j-docker played nice with this feature and supported it right out of the box. 
Also by supporting this, clients can use a common connection string (e.g. bolt+routing://neo4jcore:7687)
without much configuration/reverse proxy help.

I'm thinking if it can generate the IP address list for the option `NEO4J_causalClustering_initialDiscoveryMembers` by doing nslookup on an intermediate environment variable declared in the compose file.

**Comment by Tvaroh** [comment]
Even better would be to support dynamic (broadcast or something) initial members discovery (to free people from configuring that by hand when it's not strictly needed), something that proves to be very inconvenient on ECS.

**Comment by albert-the-creator** [comment]
@jarellm were you able to find a solution to this issue? I'm currently experimenting with swarm and running into the same problem.

**Comment by mneedham** [comment]
From the next released version >= 3.2 (i.e. 3.2.3 when it's released) we will have support for putting a DNS name as your discovery member. https://github.com/neo4j/neo4j/pull/9655

We still need to add an environment variable to the Docker entry point script so you can control the discovery mode.

**Comment by mneedham** [comment]
@jarellm I think the drivers already support DNS lookup but I'm not sure what version that was introduced. Maybe try out the 1.4 series - I'm pretty sure it's implemented in there.

**Comment by jarellm** [comment]
@albert-the-creator, I settled for the workaround I provided since we won't be scaling the service any time soon anyway. But it seems that this has been solved in v3.2 as stated by @mneedham 

hey @mneedham not sure if you're referring to the bolt+routing protocol, but I got no issues re: 
neo4j client -[bolt+routing]->causal cluster

My issue was during discovery part of the core servers which I think is solved by the pull request.  Creating a custom Docker image to add this will work for now. 

Many thanks!
